diff --git a/dist/index.cjs b/dist/index.cjs
index 91cbf849c7d96511557d726db4aeb83f6f42f057..4db027cc4054282475c1b97e5cd1fff509c462e1 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -727,7 +727,7 @@ var ReactRenderer = class {
     if (className) {
       this.element.classList.add(...className.split(" "));
     }
-    if (this.editor.isInitialized) {
+    if (this.editor.isInitialized && !isReact19Plus()) {
       (0, import_react_dom2.flushSync)(() => {
         this.render();
       });
diff --git a/dist/index.js b/dist/index.js
index bf7aa68812a0eb3a00b5bb4689d2cf5a1959ccc3..6562bb1e453b8b9530567400a2f2ab114d6d4f71 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -671,7 +671,7 @@ var ReactRenderer = class {
     if (className) {
       this.element.classList.add(...className.split(" "));
     }
-    if (this.editor.isInitialized) {
+    if (this.editor.isInitialized && !isReact19Plus()) {
       flushSync(() => {
         this.render();
       });
